src/Renderer.ts

```diff
@@ -428,26 +486,24 @@ export default class Renderer {
 
     // Enable color masking mode if set
     if (Array.isArray(colorMask))
-      this.gl.uniform4fv(this._currentShader.uniforms.u_colorMask, colorMask);
+      this.gl.uniform4fv(shader.uniforms.u_colorMask, colorMask);
 
     // Used for mapping drawn sprites back to their indices in a list.
     // By looking at the color of a given pixel, we can tell which sprite is
     // the topmost one drawn on that pixel.
-    if (drawMode === ShaderManager.DrawModes.SPRITE_ID) {
-      this.gl.uniform3fv(
-        this._currentShader.uniforms.u_spriteId,
-        idToColor(spriteColorId)
-      );
+    if (drawMode === ShaderManager.DrawModes.SPRITE_ID && spriteColorId) {
```

Comment:
TypeScript doesn't have static asserts, so we can't statically error.

The other way of doing this would be a discriminated union type (like Rust's `enum`), which TypeScript also doesn't really have.